### PR TITLE
scx_layered: add a clang fmt to see what folks think

### PR DESCRIPTION
**Pull request by likewhatevs**
add a clang fmt to see what folks think

### CODE REVIEW

**Comment by likewhatevs** [comment]
I'm not sure if this is better or worse but I do know nobody likes fixing tabs and spaces all the time so, LMK what's off with this config and I'll see if I can coax an LLM into getting us one that works.

File: scheds/rust/scx_layered/.clang-format [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+SpaceBeforeParens: ControlStatements
+SpacesInParentheses: false
+
+UseTab: ForIndentation
+IndentWidth: 4
+TabWidth: 4
+IndentCaseLabels: true
+AlignAfterOpenBracket: Align
+AllowShortFunctionsOnASingleLine: Inline
+
```
Comment (hodgesds) [inline scheds/rust/scx_layered/.clang-format:22]: Maybe try adding:
```
AlignConsecutiveAssignments: AcrossEmptyLinesAndComments
AlignConsecutiveBitFields: AcrossEmptyLinesAndComments
AlignConsecutiveMacros: AcrossEmptyLinesAndComments
```

File: scheds/rust/scx_layered/src/bpf/main.bpf.c [hunk 2]
```diff
@@ -56,8 +56,16 @@ static u32 preempt_cursor;
 
 extern unsigned CONFIG_HZ __kconfig;
 
-#define dbg(fmt, args...)	do { if (debug) bpf_printk(fmt, ##args); } while (0)
-#define trace(fmt, args...)	do { if (debug > 1) bpf_printk(fmt, ##args); } while (0)
+#define dbg(fmt, args...)            \
```
Comment (hodgesds) [inline scheds/rust/scx_layered/src/bpf/main.bpf.c:59]: No bueno on this

Comment (likewhatevs) [reply scheds/rust/scx_layered/src/bpf/main.bpf.c:59]: really? thought that was one of the nicer bits, alright will see about fixing that.

File: scheds/rust/scx_layered/src/bpf/intf.h [hunk 3]
```diff
@@ -29,34 +29,34 @@ typedef unsigned long long u64;
...
-	MIN_LAYER_WEIGHT	= 1,
-	DEFAULT_LAYER_WEIGHT	= 100,
-	USAGE_HALF_LIFE		= 100000000,	/* 100ms */
-	LAYER_LAT_DECAY_FACTOR	= 4,
-
-	HI_FALLBACK_DSQ_BASE	= MAX_LAYERS * MAX_LLCS,
-	LO_FALLBACK_DSQ		= (MAX_LAYERS * MAX_LLCS) + MAX_LLCS + 1,
+	CACHELINE_SIZE		   = 64,
+	MAX_CPUS_SHIFT		   = 9,
+	MAX_CPUS			   = 1 << MAX_CPUS_SHIFT,
```
Comment (likewhatevs) [inline scheds/rust/scx_layered/src/bpf/intf.h:34]: those 3 added some weirdness here I think @hodgesds

**Comment by likewhatevs** [comment]
Guess I'll revisit this later lol, might be beyond clang-fmt's/rando LLM's capabilities atm.

All I wanted was the things I always get wrong:
* leading tabs not spaces, preferably matching the indent level if possible.
* spaces before and after parens in conditional statements.
* brackets on newline for function's

File: scheds/rust/scx_layered/.clang-format (hunk 1 above)
Comment (hodgesds) [reply scheds/rust/scx_layered/.clang-format:25]: Hmm maybe they didn't work so well

**Pull request closed without merging**